walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,94 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
+
+  private static boolean attendanceRequired(Event event, MeetingRequest request) {
+     Set<String> s = event.getAttendees();
+     Collection<String> c = request.getAttendees();
+     for (String person : c) {
+       if (s.contains(person)){ return true; }
+     }
+     return false;
+  }
+
+  /**
+   * Returns a list of times that are compatible with a given list of events
+   * @param {Collection<Event>} events A list of events where each event contains
+   * information about when a specific group of people are meeting.
+   * @param {MeetingRequest} request An object describing the meeting being requested.
+   * @return {Collection<TimeRange>} A list of meeting times where all required 
+   * attendees are available
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Make two sorted lists, based on meeting start and end time.
+    List<TimeRange> busyTimesStart = new ArrayList<TimeRange>();
+    List<TimeRange> busyTimesEnd = new ArrayList<TimeRange>();
```

Comment:
1. It's a little bit mysterious to me why you need the two different arrays sorted in different ways. I think you're trying to do something where you are merging overlapping ```TimeRanges```?
Right now you are using two lists with two pointers. I think the approach might be bugged (not 100% sure, honestly) which I wrote about below. I think that per loop you probably also need to maintain a local variable or mutate one of the lists. But in doing that you'll probably discover a way to do this with just using one list.

2. The comment here just summarizes the next ten lines or so which I didn't really need since it's really clear what you're doing. In general comments annotating a declaration, single conditional statement or simple loop does is usually not needed. 
A comment that would be valuable here would be one that explains why I actually need to keep track of two lists at all. This was some context I was missing when I started reading the real "meat" of this function which is lines 70-90.

Reply:
yep, switching to one list